bindings/java/src/org/sleuthkit/datamodel/SleuthkitCase.java

```
@@ -13878,8 +13882,41 @@ public CaseDbConnection getPooledConnection() throws SQLException {
 					logger.log(Level.WARNING, String.format("Thread %s (ID = %d) already has an open transaction.  New connection may encounter SQLITE_BUSY error. ", Thread.currentThread().getName(), Thread.currentThread().getId()), new Throwable());
 				}
 			}
-			return new SQLiteConnection(getPooledDataSource().getConnection());
+			java.sql.Connection conn = getPooledDataSource().getConnection();
+			runOptimize(conn);
+			CaseDbConnection caseDbConn = new SQLiteConnection(conn);
+			return caseDbConn;
 		}
+
+		/**
+		 * Performs optimization based on
+		 * https://www.sqlite.org/pragma.html#pragma_optimize advice regarding
+		 * long-lived connections.
+		 */
+		private void runOptimize(java.sql.Connection conn) {
+
+			// get current time
+			long thisStart = System.currentTimeMillis();
+
+			// run optimize
+			try (Statement statement = conn.createStatement()) {
+				statement.execute("PRAGMA optimize");
```

> Applications that use long-lived database connections should run "PRAGMA optimize=0x10002;" 
Shouldn't we follow their advice here? 

>  and then also run "PRAGMA optimize;" periodically, 
perhaps we can have a [ScheduledExecutorService](https://docs.oracle.com/javase/8/docs/api/java/util/concurrent/ScheduledExecutorService.html)  that does this call every couple of hrs or so?

---

Hi @jayaramcs , my issue right now is that I don't have a way to determine when the last time `PRAGMA optimize` has been called on a connection because I haven't found a way yet to identify the physical connection from c3p0's pooled data source proxy connection wrapper.  I looked into it a little bit and there may be something in `Connection#getMetaData` that might be a clue to use.

Regarding `PRAGMA optimize = 0x10002` vs. `PRAGMA optimize`, I could be wrong, but I think `PRAGMA optimize` covers the `0x10002` mask.  At least that is what I infer from the [doc](https://www.sqlite.org/pragma.html#pragma_optimize):
> The default MASK is 0xfffe.

So, I figured it was best to play it safe and call `PRAGMA optimize` with the default mask at least until I can determine unique connections.

I think running periodically per connection is worth doing.  My concern right now is that I want to make sure that each connection gets the optimize call- for my case, certainly after ingest but before that connection is used for Hayabusa.  I'm not quite sure yet how to identify different unique physical connections to determine when the last time that connection had optimize called on it.

---

Ok. lets get this over to perf runs see how it fares.